### PR TITLE
[addons] add first gui control addon class for buttons

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description

This add the first class on new addon interface to edit a button
control.

Hold on this alone to see better how the controls becomes used
on new addon system.

On next request are all controls together.

Here the smallest example to edit a control on new way:
```cpp
// this = The addon window class
// 33 = The on skin.xml used control id
kodi::gui::control::CButton(this, 33).SetVisible(false);
```

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/addons/interfaces/GUI/Window.cpp [hunk 1]
```diff
@@ -820,6 +823,25 @@ void Interface_GUIWindow::mark_dirty_region(void* kodiBase, void* handle)
 }
 //@}
 
+/*!
+ * To get control functions
+ */
+//@{
+void* Interface_GUIWindow::getcontrol_button(void* kodiBase, void* handle, int control_id)
```
Comment (ksooo) [inline xbmc/addons/interfaces/GUI/Window.cpp]: getcontrol_button => get_control_button? (missing underscore between "get" and "control")

File: xbmc/addons/interfaces/GUI/Window.h [hunk 2]
```diff
@@ -104,6 +104,9 @@ namespace ADDON
 
     /* Various functions */
     static void mark_dirty_region(void* kodiBase, void* handle);
+
+    /* To get control functions */
```
Comment (ksooo) [inline xbmc/addons/interfaces/GUI/Window.h]: "GUI control access functions"

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/gui/ControlButton.h [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  ///
+  /// It has the header \ref ControlButton.h "#include <kodi/gui/ControlButton.h>"
+  /// be included to enjoy it.
+  ///
+  /// Here you find the needed skin part for a \ref skin_Button_control "button control"
+  ///
+  /// @note The call of the control is  only  possible  from  the  corresponding
+  /// window as its class and identification number is required.
+  ///
+  class CControlButton : public CAddonGUIControlBase
```
Comment (ksooo) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/gui/ControlButton.h]: I would not use "CControl" prefix for all classes modelling GUI controls. This is what namespaces are for. 

I would use just "CButton" here. For me it is quite obvious  that a class contained in namespace "kodi::gui" named "CButton" is a GUI control. Even better, you could put all GUI controls in namespace kodi::gui::control.

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/gui/definitions.h [hunk 4]
```diff
@@ -213,11 +223,15 @@ typedef struct AddonToKodiFuncTable_kodi_gui_window
 
   /* Various functions */
   void (*mark_dirty_region)(void* kodiBase, void* handle);
+
+  /* To get control functions */
```
Comment (ksooo) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/gui/definitions.h]: "GUI control access functions"

**Review by ksooo — CHANGES_REQUESTED** [review]
As always, some comments and suggestions.

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/gui/ControlButton.h (hunk 3 above)
Comment (AlwinEsch) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/gui/ControlButton.h]: For me not so good to have it as "CButton" alone. In this case should also dialogs renamed and in own namespace.

Further want I hold it a bit equal with the Python where it is named "ControlButton"

Comment (ksooo) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/gui/ControlButton.h]: You are the boss. At the end it is just names.

Comment (ksooo) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/gui/ControlButton.h]: I think we should use the features of the languages we support. And C++ has namespaces, a concept which fits very nicely here. The fact that it is (currently) named the same for Python is not a very strong point here. And yes, dialogs should go to kodi::gui::dialogs, then.

**Review by ksooo — APPROVED** [review]

**Comment by ksooo** [comment]
Very nice. Feel free to add another commit to change dialogs namespace (as we discussed on Slack).

**Comment by AlwinEsch** [comment]
Currently on work of them :)

**Review by ksooo — APPROVED** [review]
Good to merge, imo.

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
OSX-64 jenkins error is unrelated.

**Pull request merged**